docs/source/notes/resumption.rst

```diff
@@ -0,0 +1,134 @@
+|:play_or_pause_button:| Auto Resumption
+========================================
+
+Resuming from checkpoints is commonly used to recover from hardware failures (e.g. spot instances in the cloud being terminiated), loss spikes for large language models, or other unforseen errors. Our trainer supports resuming from checkpoints with the ``load_path`` argument (see :doc:`/trainer/checkpointing` for more details).
+
+.. testsetup::
+
+    import os
+    import shutil
+
+    from composer import Trainer
+
+    trainer = Trainer(
+        model=model,
+        train_dataloader=train_dataloader,
+        max_duration="1ep",
+        save_filename="ep{epoch}.pt",
+        save_folder="./path/to/checkpoints",
+        save_overwrite=True,
+        save_interval="1ep",  # Save checkpoints every epoch
+    )
+    trainer.fit()
+
+    assert os.path.exists("./path/to/checkpoints/ep1.pt")
+
+    if not os.path.exists("./path/to/checkpoints/ep25.pt"):
+        shutil.copy2("./path/to/checkpoints/ep1.pt", "./path/to/checkpoints/ep25.pt")
+
+    assert os.path.exists("./path/to/checkpoints/ep25.pt")
+
+
+.. testcode::
+
+    x = 2
+
+However, recovering from a failure here would still require manual intervention to relaunch a new job with the ``load_path`` pointing to the correct checkpoint.
+
+Instead, our trainer supports the ``autoresume=True`` feature. With autoresume, the trainer will automatically check the ``save_folder`` for the latest checkpoints and resume training.
+
+.. testcode::
+
+    trainer = Trainer(
+        ...,
+        autoresume=True,
+        save_folder='./path/to/folder',
+        run_name='my_cool_run',
+    )
+
+With autoresume, users can re-submit the _same_ code to the training run, and the trainer will handle finding and resuming from the latest checkpoints. This works well with systems like Kubernetes that automatically resubmit the same job when there is a node failure (due to spot instances as well). For ``autoresume=True`` to work, we require that both a ``save_folder`` and a ``run_name`` be provided. These are used to search for existing checkpoints.
+
+For an example code, see ``examples/checkpoint_autoresume.py``.
```

Comment:
(This can be done in a future PR), but I would like to convert this example script into a notebook. For now, can this be a link to the file on the dev branch?

Reply:
Done